### PR TITLE
URLSearchParams and isomorphic-fetch

### PR DESCRIPTION
**Pull request by VishnAndr**
Introduces:

- native (nodejs) URLSearchParams

- ~~dtoNames in query method is optional (by extracting required DTOs from coreSQL statement)~~

Changes:
- node-fetch to isomorphic-fetch to enable FSM SDK to be used in the browser (for example, with [ShellSDK](https://github.com/SAP/fsm-shell))

### CODE REVIEW

File: src/core-api.client.ts [hunk 1]
```diff
@@ -182,12 +177,12 @@ export class CoreAPIClient {
...
    */
-  public async query<T extends { [projection: string]: DTOModels }>(coreSQL: string, dtoNames: DTOName[]): Promise<{ data: T[] }> {
+  public async query<T extends { [projection: string]: DTOModels }>(coreSQL: string, dtoNames?: DTOName[]): Promise<{ data: T[] }> {
     const token = await this._ensureToken();
-    const params = new URLSearchParams({
+    const params = new URLSearchParams(Object.assign({ 
       ...RequestOptionsFacory._getRequestAccountQueryParams(token, this._config),
-      dtos: RequestOptionsFacory.getDTOVersionsString(dtoNames)
-    });
+      dtos: RequestOptionsFacory.getDTOVersionsString(dtoNames || [... new Set(coreSQL.match(/(?<=from\s*|join\s*)(\w+)/gi))] as DTOName[]),
```
Comment (GauSim) [inline src/core-api.client.ts:184]: There are possible cases where this fails with grouping and sub in queries.
For usability, we should keep this explicit on the public API of this client - to keep the magic out and easy understandable. 
`DTOName` type helps in the IDE with auto complete also the dirty type cast then is not needed.
You can of cause still do regexing when calling the client externally. Please revert.

File: src/core-api.client.ts [hunk 2]
```diff
@@ -267,7 +262,7 @@ export class CoreAPIClient {
       clientIdentifier: this._config.clientIdentifier,
       dtos: RequestOptionsFacory.getDTOVersionsString(actions.map(it => it.dtoName)),
       ...RequestOptionsFacory._getRequestAccountQueryParams(token, this._config),
-    }));
+    }) as { [key: string]: string });
```
Comment (GauSim) [inline src/core-api.client.ts:265]: Please revert to type inference

File: src/core-api.client.ts [hunk 3]
```diff
@@ -182,12 +177,12 @@ export class CoreAPIClient {
    * @param coreSQL
    * @param dtoNames
    */
-  public async query<T extends { [projection: string]: DTOModels }>(coreSQL: string, dtoNames: DTOName[]): Promise<{ data: T[] }> {
+  public async query<T extends { [projection: string]: DTOModels }>(coreSQL: string, dtoNames?: DTOName[]): Promise<{ data: T[] }> {
```
Comment (GauSim) [inline src/core-api.client.ts:180]: see below - please to not change the public interface

**Review by GauSim — CHANGES_REQUESTED** [review]
See comments on changing the public interface, please revert the change.
please also update [CHANGELOG](https://github.com/SAP/fsm-sdk/blob/master/CHANGELOG.md) and bump version.

File: src/core-api.client.ts (hunk 1 above)
Comment (VishnAndr) [reply src/core-api.client.ts:184]: Some possible cases examples would be nice to see. Maybe they can be addressed by adjusting the regex.

**Review by VishnAndr — COMMENTED** [review]
Reverted query changes.

**Comment by GauSim** [comment]
Closing in favor of https://github.com/SAP/fsm-sdk/pull/10

**Pull request closed without merging**